modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableRowSkinTest.java

```diff
@@ -0,0 +1,125 @@
+
+    @Before
+    public void before() {
+        tableView = new TableView<>();
+
+        TableColumn<Person, String> firstNameCol = new TableColumn<>("Firstname");
+        firstNameCol.setCellValueFactory(new PropertyValueFactory<>("firstName"));
+        TableColumn<Person, String> lastNameCol = new TableColumn<>("Lastname");
+        lastNameCol.setCellValueFactory(new PropertyValueFactory<>("lastName"));
+        TableColumn<Person, String> emailCol = new TableColumn<>("Email");
+        emailCol.setCellValueFactory(new PropertyValueFactory<>("email"));
+        TableColumn<Person, Integer> ageCol = new TableColumn<>("Age");
+        ageCol.setCellValueFactory(new PropertyValueFactory<>("age"));
+
+        tableView.getColumns().addAll(firstNameCol, lastNameCol, emailCol, ageCol);
+
+        ObservableList<Person> items = FXCollections.observableArrayList(
+                new Person("firstName1", "lastName1", "email1@javafx.com", 1),
+                new Person("firstName2", "lastName2", "email2@javafx.com", 2),
+                new Person("firstName3", "lastName3", "email3@javafx.com", 3),
+                new Person("firstName4", "lastName4", "email4@javafx.com", 4)
+        );
+
+        tableView.setItems(items);
+
+        stageLoader = new StageLoader(tableView);
+    }
+
+    @Test
+    public void removedColumnsShouldRemoveCorrespondingCellsInRowFixedCellSize() {
+        tableView.setFixedCellSize(24);
+        removedColumnsShouldRemoveCorrespondingCellsInRowImpl();
+    }
+
+    @Test
+    public void removedColumnsShouldRemoveCorrespondingCellsInRow() {
+        removedColumnsShouldRemoveCorrespondingCellsInRowImpl();
+    }
+
+    @Test
+    public void invisibleColumnsShouldRemoveCorrespondingCellsInRowFixedCellSize() {
+        tableView.setFixedCellSize(24);
+        invisibleColumnsShouldRemoveCorrespondingCellsInRowImpl();
+    }
+
+    @Test
+    public void invisibleColumnsShouldRemoveCorrespondingCellsInRow() {
+        invisibleColumnsShouldRemoveCorrespondingCellsInRowImpl();
+    }
+
+    @After
+    public void after() {
+        stageLoader.dispose();
+    }
+
+    private void invisibleColumnsShouldRemoveCorrespondingCellsInRowImpl() {
+        // Set the last 2 columns invisible.
+        tableView.getColumns().get(tableView.getColumns().size() - 1).setVisible(false);
+        tableView.getColumns().get(tableView.getColumns().size() - 2).setVisible(false);
+
+        Toolkit.getToolkit().firePulse();
+
+        // We set 2 columns to invisible, so the cell count should be decremented by 2 as well.
+        assertEquals(tableView.getColumns().size() - 2,
+                VirtualFlowTestUtils.getCell(tableView, 0).getChildrenUnmodifiable().size());
+    }
+
+    private void removedColumnsShouldRemoveCorrespondingCellsInRowImpl() {
+        // Remove the last 2 columns.
+        tableView.getColumns().remove(tableView.getColumns().size() - 2, tableView.getColumns().size() - 1);
```

Comment:
code comment is not correct - the last parameter of `remove(int, int)` is exclusive. While it doesn't change the outcome (failing/passing before/after fix), it might confuse future readers :)

Reply:
Thanks, didn't recognized that. I will change it so that **really** the last 2 columns are removed.